bundles/http/Citrix/Cherwell/Cherwell.js

```diff
@@ -731,31 +734,43 @@ async function getRequest({ dataStore, client }, moduleName, body, endpoint) {
         const data = await resp.json();
         const dataArray = data.records.map(obj => {
             let rObj = JSON.parse(obj);
-            rObj.CreatedDateTime = new Date(obj.CreatedDateTime);
-            rObj.LastModifiedDateTime = new Date(obj.LastModifiedDateTime);
-            rObj.LastModDateTime = new Date(obj.LastModDateTime);
-            rObj.SLAResponseWarning = new Date(obj.SLAResponseWarning);
-            rObj.SLAResolutionWarning = new Date(obj.SLAResolutionWarning);
+            rObj.CreatedDateTime = new Date(rObj.CreatedDateTime);
+            rObj.LastModifiedDateTime = new Date(rObj.LastModifiedDateTime);
+            rObj.LastModDateTime = new Date(rObj.LastModDateTime);
+            rObj.SLAResponseWarning = new Date(rObj.SLAResponseWarning);
+            rObj.SLAResolutionWarning = new Date(rObj.SLAResolutionWarning);
             return rObj;
         });
 
         if (moduleName == 'servicesIncidentTypes' || moduleName == 'categoriesIncidentTypes') {
-            let types = dataArray.map(obj => {
+            let types = dataArray.map((obj, index) => {
                 const rType = {
+                    id: index,
                     Service: obj.Service,
                     IncidentType: obj.IncidentType,
                     Category: obj.Category
                 };
                 return rType;
             });;
 
+            let uniqTypes = [];
             if (moduleName == 'categoriesIncidentTypes') {
-                types = _.uniqBy(types, 'Category');
+                for (const element of types) {
+                    let contains = await listContains(uniqTypes, element, 'Category', 'IncidentType');
+                    if (!contains) {
+                        uniqTypes.push(element);
+                    }
+                }
             } else {
-                types = _.uniqBy(types, 'Service');
```

Comment:
sorry I probably confused you with `_.uniqBy`, maybe you would use [_.uniqWith](https://stackoverflow.com/questions/40381771/lodash-uniqwith-with-an-array-of-objects#answer-40382536)

Reply:
uniqWith function implemented